### PR TITLE
Add IDOBAO ID67

### PR DESCRIPTION
**Pull request by mehoron**
Add the IDOBAO ID67 Keybaord

## Description

This adds the IDOBAO ID67 keyboard.
It includes:
 * Layout
 * Default Keymap
 * VIA Keymap

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by sigprof — CHANGES_REQUESTED** [review]
This code looks much like the output of `kbfirmware.com` (or some of its clones) — it uses obsolete functions, `KEYMAP()` instead of `LAYOUT()`, obsolete variables and comments in `rules.mk`, etc. If this is actually the case, instead of fixing all those issues manually, it would be much easier to redo the conversion using https://noroadsleft.github.io/kbf_qmk_converter/ — that converter generates files which actually use modern QMK conventions (and also formats the keymaps in a nicer way).

**Comment by mehoron** [comment]
@sigprof Thanks for taking a look, I was indeed using some old keyboards as references. I've gone ahead and revamped everything to your suggestions as well as removed some empty functions that were bothering me. I compiled both keymaps and had the results tested as well.

File: keyboards/id67/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "id67.h"
```
Comment (drashna) [inline keyboards/id67/keymaps/via/keymap.c:17]:
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/id67/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "id67.h"
```
Comment (drashna) [inline keyboards/id67/keymaps/default/keymap.c:17]:
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/id67/readme.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+## ANSI support:
+
+* Keyboard Maintainer: Tybera
+* Hardware Supported: IDOBAO ID67
+* Hardware Availability: [IDOBAO]https://www.idobao.net/products/idobao-id67-65-hot-swappable-mechanical-keyboard-kit-1
+
+Make example for this keyboard (after setting up your build environment):
+
+    make id67
```
Comment (drashna) [inline keyboards/id67/readme.md:13]:
```suggestion
Make example for this keyboard (after setting up your build environment):

    make id67:default

Flashing example for this keyboard:

    make id67:default:flash


See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).

```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by mehoron** [comment]
Thanks for taking a look @drashna I've committed your suggestions and made note of them for future projects.

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**